### DISCUSSION TITLE
TLS error with the specific host

### DISCUSSION THREAD

**Discussion by szelga** [ipxe/ipxe#660] [General]
tried to chainload from the host https://netbox.pssrv.ru/ , got an expired certificate error:

the log goes like this:

```
> chain https://netboot.pssrv.ru/netboot
https://netboot.pssrv.ru/netboot...X509 chain 0xed484 added X509 0xfa624 "netboot.pssrv.ru"
X509 chain 0xed484 added X509 0xfca44 "R3"
X509 chain 0xed484 added X509 0xfd024 "ISRG Root X1"
X509 chain 0xed484 added X509 0xf2ab4 "DST Root CA X3"
X509 chain 0xed484 added X509 0xf7e24 "iPXE cross-signing CA"
X509 chain 0xed484 added X509 0xf8244 "iPXE root CA"
X509 0xf2ab4 "DST Root CA X3" has expired (at time 1654597353)
X509 0xf2ab4 "DST Root CA X3" has expired (at time 1654597353)
Permission denied (https://ipxe.org/0216e43c)
```

it seems, the error is rooted in expired iPXE CA. I tried embedding CA from Firefox and building like that, but the result remained the same. what am I doing wrong?

**Comment by stappersg** [comment]
```text
$ curl --verbose https://netboot.pssrv.ru/netboot
*   Trying 80.85.152.209:443...
* Connected to netboot.pssrv.ru (80.85.152.209) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS header, Finished (20):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS header, Finished (20):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server accepted http/1.1
* Server certificate:
*  subject: CN=netboot.pssrv.ru
*  start date: Jun  4 21:27:47 2022 GMT
*  expire date: Sep  2 21:27:46 2022 GMT
*  subjectAltName: host "netboot.pssrv.ru" matched cert's "netboot.pssrv.ru"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
> GET /netboot HTTP/1.1
> Host: netboot.pssrv.ru
> User-Agent: curl/7.83.1
> Accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Server: nginx/1.21.6
< Date: Tue, 07 Jun 2022 19:59:54 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 32
< Connection: keep-alive
< 
* Connection #0 to host netboot.pssrv.ru left intact
Required field 'uuid' is missing
```
The `start date: Jun  4 21:27:47 2022 GMT` made me think that webserver wasn't yet reloaded on _Jun  7_, day of this discussion was started.

Advice / request to @szelga _try again_.

**Comment by szelga** [comment]
just to be clear, are you suggesting to regenerate the certificate?

**Reply by stappersg** [reply]
No.  I'm suggesting to try the netboot where https is involded again.   Because I see a fresh certificate and have expirence first hand that certificates where refreshed but the webserver wasn't yet exposing the cert.

I'm saying that I do **not** share the _"it seems, the error is rooted in expired iPXE CA"_

**Reply by szelga** [reply]
still the same after restarting the web server. also, the certificate was created a few days ago from scratch, I think.

**Reply by stappersg** [reply]
Oops.

I think that @szelga is saying something like _"I'm lost,  all this encryption stuff does not bring me what I hoped for, please help me."_

I hope that reports like _"iPXE talking HTTPS to server works for me"_ do show up. Yes, I'm saying that my use case of iPXE doesn't use HTTPS.

**Reply by szelga** [reply]
I understood from your post that you've never used HTTPS with iPXE and don't know how to debug it, right?

**Comment by stappersg** [comment]
On Wed, Jun 08, 2022 at 04:39:14AM -0700, Wasil Siargiejczyk wrote:
> I understood from your post that you've never used HTTPS with iPXE and don't know how to debug it, right?

Yes, maybe.

And yes, Wasil understood correctly that this problem is not my problem.


Groeten
Geert Stappers
-- 
Silence is hard to parse
